### PR TITLE
Add a swift example

### LINKED ISSUE #435: Add a swift example
Swift support ([almost](https://github.com/dart-lang/ffigen/issues/434)) already works, building on ObjC. Add an example.

Swift command to generate the dylib and objc wrapper header:

```
swiftc -c foo.swift -emit-objc-header-path foo.h -emit-library
```

Then we can run ffigen on foo.h to get the Dart bindings.

### PR DESCRIPTION
**Pull request by liamappelbe**
The example doesn't do anything exciting, just interacts with a method and a field on a Swift class.

I wrote a test for this example, but it's blocked on #443. I'll add it when I fix that bug, but this example needs to land this week.

Fixes #435

### CODE REVIEW

File: example/swift/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  objc-interfaces:
+    include:
+      - 'SwiftClass'
+    module:
+      'SwiftClass': 'swift_module'
+  headers:
+    entry-points:
+      - 'swift_api.h'
+  preamble: |
+    // ignore_for_file: camel_case_types, non_constant_identifier_names, unused_element, unused_field, return_of_invalid_type, void_checks, annotate_overrides, no_leading_underscores_for_local_identifiers, library_private_types_in_public_api
```
Comment (dcharkes) [inline example/swift/pubspec.yaml:32]: nit: This leads to more than 80 chars per line here and in the generated file. Consider making it multiple lines.

File: example/swift/example.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'swift_api_bindings.dart';
+
+void main() {
```
Comment (dcharkes) [inline example/swift/example.dart:8]: Please add a TODO to run this as test on the CI with a reference to the blocking issue.

**Review by dcharkes — APPROVED** [review]
lgtm

> I wrote a test for this example, but it's blocked on https://github.com/dart-lang/ffigen/issues/443. I'll add it when I fix that bug,

I have a slight preference for fixing that bug first so that we only land tested code, but feel free to land this PR with a TODO referring to that bug.

> but this example needs to land this week.

Where does this constraint come from?

**Comment by liamappelbe** [comment]
> Where does this constraint come from?

The blog post for Dart 2.18 is going to mention ObjC and Swift interop. Michael is working on the blog post now, and I have a couple of things to land in relation to that, including this. Landing it last week wasn't a strict deadline. Landing it today is fine.

**Pull request merged**